modules/opentelemetry-common/src/main/scala/io/janstenpickle/trace4cats/opentelemetry/common/Trace4CatsSpanData.scala

```diff
@@ -22,15 +22,19 @@ object Trace4CatsSpanData {
 
       override lazy val getSpanId: String = span.context.spanId.show
 
-      override lazy val isSampled: Boolean = span.context.traceFlags.sampled.toBoolean
+      private lazy val getTraceFlags: TraceFlags =
+        if (span.context.traceFlags.sampled.toBoolean) TraceFlags.getSampled else TraceFlags.getDefault
```

Comment:
@janstenpickle this mapping between `SampleDecision` ADT and otel `isSampled` flag is weird. Please look at this carefully. Does `Drop` really match to `true` (is sampled)?

I remember there was an issue with W3C flags encoding. I hope this is not a similar one?

Reply:
I also a bit worried about the parent SpanContext's and Links' trace flags in the same file. We used `TraceFlags.getSampled`, not `TraceFlags.getDefault`. Still not sure why.